internal/discovery/discovery.go

```diff
@@ -69,3 +78,73 @@ func Open(sourceDir string) ([]byte, error) {
 
 	return c, nil
 }
+
+func GetDiscoveryFileLastChanged(sourceDir string) (time.Time, error) {
```

Comment:
I don't think it is useful to scope this function to the discoveryFile. It could just be: `GetFileLastChanged(name string)`